src/python/WMCore/Services/DBS/DBS3Reader.py

```diff
@@ -307,6 +310,25 @@ def listDatasetFileDetails(self, datasetPath, getParents=False, getLumis=True, v
             files[f['logical_file_name']]['Lumis'] = {}
             files[f['logical_file_name']]['Parents'] = []
 
+        # parallel execution for listFileParents and listFileLumis APIs
+        if self.parallel:
```

Comment:
Your changes do two things:
- they modify existing code base, since you add new loop to create block_parents dict
- they create new dict (which adds to the memory consumption)

In my version, I did not change existing codebase.

It is a choice and I don't have strong preference in one way or another. I leave it up to you to decide. Bottom line, the concurrent/parallel construct in a code quite often require new redesign and it is hard to align them with sequential programming patterns.

Reply:
The memory concern is a good point. However, if we use the concurrent version, then we will end up with that large dictionary anyways.

As anything, there are pros and cons and it's hard to decide whether a) making a transparent change now is better than b) making the code maintenance easier for future changes. Given that you have already coded and tested it, I'd say we keep the current implementation.